### ISSUE TITLE
upb/port_def.inc issue on electron-rebuild

### ISSUE THREAD

**Issue by eouia** [grpc/grpc-node#1057]
### Problem description
electron-rebuild failed with error;

### Reproduction steps
```
npm install electron-rebuild
./node_modules/.bin/electron-rebuild
```

### Environment
 - OS name, version and architecture: Raspbian Buster (Raspberry Pi 4B)
 - Node version : 10.16.0
 - Node installation method : manual
 - If applicable, compiler version : unknown
 - Package name and version : 1.14.0 (not sure)

### Error
```
CXX(target) Release/obj.target/grpc/deps/grpc/src/core/lib/security/credentials/alts/alts_credentials.o
In file included from ../deps/grpc/third_party/upb/upb/msg.h:14,
                 from ../deps/grpc/third_party/upb/upb/generated_util.h:10,
                 from ../deps/grpc/src/core/ext/upb-generated/src/proto/grpc/gcp/transport_security_common.upb.h:12,
                 from ../deps/grpc/src/core/tsi/alts/handshaker/transport_security_common_api.h:29,
                 from ../deps/grpc/src/core/lib/security/credentials/alts/grpc_alts_credentials_options.h:26,
                 from ../deps/grpc/src/core/lib/security/credentials/alts/alts_credentials.h:26,
                 from ../deps/grpc/src/core/lib/security/credentials/alts/alts_credentials.cc:21:
../deps/grpc/third_party/upb/upb/upb.h:27:10: fatal error: upb/port_def.inc: No such file or directory
 #include "upb/port_def.inc"
          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make: *** [grpc.target.mk:531: Release/obj.target/grpc/deps/grpc/src/core/lib/security/credentials/alts/alts_credentials.o] Error 1
make: Leaving directory '/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Linux 4.19.57-v7l+
gyp ERR! command "/usr/bin/node" "/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/.bin/node-gyp" "rebuild" "--target=3.1.10" "--arch=arm" "--dist-url=https://electronjs.org/headers" "--build-from-source" "--module_name=grpc_node" "--module_path=/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc/src/node/extension_binary/electron-v3.1-linux-arm-glibc" "--host=https://node-precompiled-binaries.grpc.io/" "--remote_path={name}/v1.24.0" "--package_name=electron-v3.1-linux-arm-glibc.tar.gz"
gyp ERR! cwd /home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/grpc
gyp ERR! node -v v10.16.0
gyp ERR! node-gyp -v v5.0.4
gyp ERR! not ok 
```

**Comment by murgatroid99** [comment]
First, thank you for reporting this. This is a packaging error that I will try to correct soon.

Second, we recommend not using `electron-rebuild` with grpc because it has problems like this. Instead please follow the instructions in the [Electron section of the gRPC readme](https://www.npmjs.com/package/grpc#about-electron).

**Comment by eouia** [comment]
Thanks I'll try it.

**Comment by eouia** [comment]
Sorry, your instruction make same errors.

**Comment by murgatroid99** [comment]
I just noticed that you said you are using version 1.14 of the library. That is an old version, and it does not support Electron 3. I suggest updating to a more recent version of the library.

**Comment by eouia** [comment]
sorry, grpc version was 1.24.0 (I think this is the newest.) Anyway, I'll look inside more.

**Comment by eouia** [comment]
I reinstall grpc@1.23.3 instead 1.24. It works. Thanks.

**Comment by murgatroid99** [comment]
If you still have it, would you mind sharing the first several lines of the install log that ended with errors?

**Comment by zikotrance** [comment]
Hi eouia, do you accept assisting me fixing the electron issue for a compensation? I've been struggled with it for a while
Please
Thanks

**Comment by zikotrance** [comment]
Please email me on malak1987@live.fr Thanks a lot :)

**Comment by murgatroid99** [comment]
@zikotrance If you have a problem that is not resolved by this issue I suggest opening a new issue describing the problem you are experiencing.

**Comment by abhatikar** [comment]
diff --git a/package.json b/package.json
index 2ce0f71..63a7097 100644
--- a/package.json
+++ b/package.json
@@ -24,9 +24,10 @@
   "dependencies": {
     "google-assistant": "^0.5.2",
     "node-record-lpcm16": "^0.3.0",
+    "grpc": "1.23.3",
     "play-sound": "^1.1.2"
   },
   "devDependencies": {
-    "electron-rebuild": "^1.8.2"
+    "electron-rebuild": "^1.8.6"
   }
 }

**Comment by murgatroid99** [comment]
I have published grpc 1.24.1 with the fix for this.
